content/docs/atom-guide/look-dev/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,725 @@
+Now, let's begin editing the code to add wind. 
+
+1. Open `{your-project-path}\Materials\Types\VegetationBending_Common.azsli`.
+   - Because we need to have the bending be done on all passes, we can add the code in the common `.azsli` file to avoid repeat code. The functions we write will be called by the vertex shaders.
+2. At the bottom, add the following code:
+
+   ```hlsl
+   float4 SetUpWindBending(float currentTime, float4 worldPosition) 
+   {
+      float2 wind = float2(MaterialSrg::m_windX, MaterialSrg::m_windY);
+      float bendingStrength = MaterialSrg::m_bendingStrength;
+      float2 amplitude = float2(wind.x * 0.4 + wind.y * 0.2, wind.y * 0.4 - wind.x * 0.2);
+      float2 frequency = float2(MaterialSrg::m_windBendingFrequency, MaterialSrg::m_windBendingFrequency * 1.125);
+      // Using the world position to modify the phase makes it so different trees near each other are at similar but not equal points in the animation, 
+      // so they appear to be reacting to the same wind but at different times as the wind moves through the vegetation.
+      float2 phase = float2(worldPosition.x * 0.08, worldPosition.y * 0.08);
+
+      float2 addBending = float2(sin(currentTime * frequency.x + phase.x) * amplitude.x, sin(currentTime * frequency.y + phase.y) * amplitude.y);
+
+      float4 result;
+      result.x = addBending.x + wind.x;
+      result.y = addBending.y + wind.y;
+      result.z = sqrt(wind.x * wind.x + wind.y * wind.y);
+      result *= bendingStrength * 0.08;
+      float2 totalBending = addBending + wind;
+      result.w = sqrt(totalBending.x * totalBending.x + totalBending.y * totalBending.y) * 0.3;
+
+      return result;
+   }
+   ```
+   Most of the code just takes the parameters that we enter in and calculates the amplitude, frequency, and phase of the movement according to the time. This is then used to calculate the appropriate movement of the vertex. 
+   Note that we also take into account the `worldPosition` for the phase. This is because we want to mimic how wind should affect nearby objects similarly but faraway objects differently. Faraway objects may not be affected by the same "breeze".
+   {{< note >}}
+   This function could potentially be run once per-object on the CPU with the results updated each frame in the ObjectSrg instead of being run once per-vertex on the GPU. It's a tradeoff between doing the computation once per frame per vegetation object + an extra SRG compile per frame per vegetation object vs. re-doing the computation on the GPU per frame per vegetation object per vertex. It's content specific, with the redundant GPU cost increasing as vertex density increases, and also depends on if the GPU is the bottleneck, and whether or not the vertex shader is the bottleneck, and also if the vertex shader is bandwidth bound or ALU bound. 
+   {{< /note >}}
+3. Now, let's call the function in our `ProceessBending` function. Inside the conditional, add a call to our wind function:
```

Comment:
The ProcessBending function as shown above is already calling SetUpWindBending, so this is a little confusing.